src/features/GetCommands.ts

```diff
@@ -0,0 +1,105 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+import * as vscode from "vscode";
+import { LanguageClient, RequestType } from "vscode-languageclient";
+import { IFeature } from "../feature";
+
+export const GetAllCommandsRequestType = new RequestType<any, any, void, void>("powerShell/getAllCommands");
+
+export class GetCommandsFeature implements IFeature {
+    private command: vscode.Disposable;
+    private languageClient: LanguageClient;
+    private commandsExplorerProvider: CommandsExplorerProvider;
+
+    constructor() {
+        this.command = vscode.commands.registerCommand("PowerShell.RefreshCommandsExplorer", () => {
+            if (this.languageClient === undefined) {
+                // TODO: Log error message
+                return;
+            }
+            this.languageClient.sendRequest(GetAllCommandsRequestType, "").then((result) => {
```

Comment:
the network call is in the anonymous function - not the constructor

Reply:
As @tylerl0706 mentioned, the network call is in the anonymous function defined on line 19.

Do we still want to consider a later refactor of it? I can add a TODO comment if we'd like to do that. Otherwise we can resolve the conversation and carry on.

Reply:
Ah, I see now. Well it might be worth spinning it out into a non-anonymous function — keeps the code out of the constructor and prevents silly people like me from misreading it.

Reply:
Spun this and the InsertCommand anonymous function into named functions 😄